server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -144,6 +144,9 @@ public void afterRefresh(boolean didRefresh) {
                                         .filter(file -> !localSegmentsPostRefresh.contains(file))
                                         .collect(Collectors.toSet())
                                         .forEach(localSegmentChecksumMap::remove);
+                                    final long lastRefreshedCheckpoint = ((InternalEngine) indexShard.getEngine())
+                                        .lastRefreshedCheckpoint();
+                                    indexShard.getEngine().translogManager().minSeqNoRequired(lastRefreshedCheckpoint + 1);
```

Comment:
Lets see if there is a way to directly access the lastRefreshedCheckpoint in the `getMinReferencedGen()`?

Reply:
There might be a way , but it would not be a right number to reference as explained above.